app/src/androidTest/java/ch/epfl/skysync/components/ConfirmFlightUITest.kt

```diff
@@ -201,25 +202,42 @@ class ConfirmFlightUITest {
   @Test
   fun confirmButtonWorksWhenDismiss() = runTest {
     composeTestRule.onNodeWithTag("AlertDialog").assertIsNotDisplayed()
+
+    composeTestRule.onNodeWithTag("LazyList").performScrollToNode(hasText("Enter Remark"))
+    composeTestRule.onNodeWithText("Confirm").assertDoesNotExist()
+
+    val setTime = composeTestRule.onAllNodesWithText("Set Time")
+    for (i in 0 until setTime.fetchSemanticsNodes().size) {
+      setTime[i].performClick()
+    }
+
     composeTestRule.onNodeWithTag("LazyList").performScrollToNode(hasText("Confirm"))
     composeTestRule.onNodeWithText("Confirm").performClick()
     composeTestRule.onNodeWithTag("AlertDialog").assertIsDisplayed()
     composeTestRule.onNodeWithTag("AlertDialogDismiss").performClick()
     composeTestRule.onNodeWithTag("AlertDialog").assertIsNotDisplayed()
 
-    // Test that navigate was not called (the route is not meaningful)
-    verify(exactly = 0) { navController.navigate(Route.MAIN) }
+    verify(exactly = 0) { navController.navigate(Route.HOME) }
   }
 
   @Test
   fun confirmButtonWorksWhenConfirm() = runTest {
```

Comment:
A bit confusing with the precedent function as they lookalike except for one word

Reply:
I agree that the difference is small. I will merge them in 1 function where I check first the dismiss button and then confirm